src/overlays/actors/ovl_En_Bat/z_en_bat.c

```diff
@@ -33,99 +40,522 @@ const ActorInit En_Bat_InitVars = {
     ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),
     ICHAIN_F32_DIV1000(gravity, -500, ICHAIN_CONTINUE),
     ICHAIN_F32(targetArrowOffset, 2000, ICHAIN_STOP),
 };
 
-#endif
+static Gfx* sWingsDLs[] = {
+    gBadBatWingsFrame0DL, gBadBatWingsFrame1DL, gBadBatWingsFrame2DL, gBadBatWingsFrame3DL, gBadBatWingsFrame4DL,
+    gBadBatWingsFrame5DL, gBadBatWingsFrame6DL, gBadBatWingsFrame7DL, gBadBatWingsFrame8DL,
+};
+
+#define BAD_BAT_MAX_NUMBER_ATTACKING 3
+
+s32 sNumberAttacking; //!< Limit number attacking player to at most `BAD_BAT_MAX_NUMBER_ATTACKING`
+s32 sAlreadySpawned;  //!< used for those spawned with room -1 in Graveyard to avoid respawn on room change
+
+void EnBat_Init(Actor* thisx, PlayState* play) {
+    EnBat* this = THIS;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    Collider_InitAndSetSphere(play, &this->collider, thisx, &sSphereInit);
+    this->collider.dim.worldSphere.radius = sSphereInit.dim.modelSphere.radius;
+    CollisionCheck_SetInfo(&thisx->colChkInfo, &sDamageTable, &sColChkInfoInit);
+    ActorShape_Init(&thisx->shape, 2000.0f, ActorShadow_DrawCircle, 25.0f);
+
+    this->animationFrame = Rand_ZeroOne() * 9.0f;
+
+    this->paramFlags = BAD_BAT_GET_PARAMFLAGS(thisx);
+    this->switchFlag = BAD_BAT_GET_SWITCHFLAG(thisx);
+    thisx->params = BAD_BAT_GET_TYPE(thisx);
+
+    thisx->depthInWater = -32000.0f;
+    Actor_SetFocus(thisx, 20.0f);
+
+    if (sAlreadySpawned) {
+        Actor_MarkForDeath(thisx);
+    } else if (EnBat_IsGraveyardOnSecondDay(play)) {
+        if (Flags_GetSwitch(play, this->switchFlag)) {
+            Actor_MarkForDeath(thisx);
+        } else {
+            this->actor.room = -1;
+        }
+    }
+
+    if (this->paramFlags & BAD_BAT_PARAMFLAG_PERCH) {
+        thisx->params = 0;
+        EnBat_SetupPerch(this);
+    } else {
+        if (thisx->params == 0x1F) {
+            thisx->params = 0;
+        }
+        EnBat_SetupFlyIdle(this);
+        while (BAD_BAT_GET_NUMBER_TO_SPAWN(thisx) > 1) {
+            Actor_SpawnAsChildAndCutscene(
+                &play->actorCtx, play, ACTOR_EN_BAT, thisx->world.pos.x + randPlusMinusPoint5Scaled(200.0f),
+                thisx->world.pos.y + randPlusMinusPoint5Scaled(100.0f),
+                thisx->world.pos.z + randPlusMinusPoint5Scaled(200.0f), randPlusMinusPoint5Scaled((f32)0x2000),
+                (f32)0xFFFF * Rand_ZeroOne(), 0, BAD_BAT_PARAMS(this->switchFlag, this->paramFlags, 0), -1,
+                thisx->unk20, NULL);
+            BAD_BAT_GET_NUMBER_TO_SPAWN(thisx)--;
+        }
+    }
+}
+
+void EnBat_Destroy(Actor* thisx, PlayState* play) {
+    EnBat* this = THIS;
+
+    Collider_DestroySphere(play, &this->collider);
+}
+
+s32 EnBat_IsGraveyardOnSecondDay(PlayState* play) {
+    if ((CURRENT_DAY == 2) && (play->sceneNum == SCENE_BOTI)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void EnBat_StepAnimation(EnBat* this, s32 frameStep) {
+    s32 previousFrame = this->animationFrame;
```

Comment:
```suggestion
    s32 prevFrame = this->animationFrame;
```
I feel we never spell out the full word `previous`